src/common/stringutils.cpp

```diff
@@ -168,30 +223,146 @@ std::wstring StrUtils::Utf8StringToUnicode(const std::string &str)
         return 0;
-
     if ((str[pos] & 0x80) == 0)
         return 1;
-    else if ((str[pos] & 0xC0) == 0xC0)
+    //nota:  quick header check without checking inside.
+    if ((str[pos] & 0xE0) == 0xC0)
         return 2;
-    else
+    if ((str[pos] & 0xF0) == 0xE0)
         return 3;
-
-    return 0;
+    if ((str[pos] & 0xF8) == 0xF0)
+        return 4;
+    /* //theorical chars, non respecting current UTF8 std - let for possible extensions
+    if ((str[pos] & 0xFC) == 0xF8)
+        // GetLogger()->Warn("UTF8 input with 5 bytes - UTF8 ?\n");
+        return 5;
+    if ((str[pos] & 0xFE) == 0xFC)
+        // GetLogger()->Warn("UTF8 input with 6 bytes - UTF8 ?\n");
+        return 6; */
+    // rescue in case of bad input placement (pos "inside" a char), return a way to get next one
+    unsigned short ret=1;
+    while (pos+ret<size && 0x80==(str[pos+ret] & 0xC0) )
+        ++ret;
+    for(unsigned short i=0;i<ret && pos+i<size;++i)
+        GetLogger()->Warn("Bad UTF8 input => try rescue: size : %d : at %d: <%c>(%#02X)\n", ret, pos+i, str[pos+i], static_cast<unsigned char>(str[pos+i]));
+    return ret; //nota : occurrences !! - ret:1 on helpers screen (sample earth level 3 fr)
 }
 
 std::size_t StrUtils::Utf8StringLength(const std::string &str)
 {
     std::size_t result = 0;
-    unsigned int i = 0;
+    std::size_t i = 0;
+    unsigned short tmp;
     while (i < str.size())
     {
-        i += Utf8CharSizeAt(str, i);
+        tmp = Utf8CharSizeAt(str, i);
+        if(!tmp)
+        {
+            GetLogger()->Warn("Bad UTF8 input : <%s> a 0 size elt at : %d, nb %d\n", str.c_str(),i,result);
+            break;
+        }
+        i+=tmp;
         ++result;
     }
     return result;
 }
 
+/**
+ * @brief      warning log for utf8 char - trace function to display eventual
+ *             problematic chars
+ * @note       don't forget to check if the wanted utf8 char is permit by the
+ *             used font ( check: http://mathew-kurian.github.io/CharacterMap/ )
+ *
+ * @param[in]  invit    The invit message
+ * @param[in]  str      input string
+ * @param[in]  pos      The position into input string
+ * @param[in]  size     preknown size of str (avoid new strlen)
+ * @param[in]  bLogStd  do log standard one chars (default false)
+ */
+void StrUtils::Utf8Log(const char*const invit, const char*const str, const std::size_t pos,
+                       const std::size_t size, const bool bLogStd)
+{
+    assert(pos<size);
+    const unsigned char len = Utf8CharSizeAt(str, pos);
+    if(1==len)
+    {
+        if(bLogStd)
+            GetLogger()->Warn("%s len %d <%c>\n", invit, len, str[pos]);
+    }
+    else
+    {
+        if(7<=len)   //improbable secu
+            GetLogger()->Warn("%s Bad UTF8 input at 1st char there : <%s> - => try rescue : %d\n",
+                invit, str+pos, len);
+        else
+        {
+            char plomp[7];
+            std::memset(plomp,0,9);
```

Comment:
I got a compile error here.
```
writing 9 bytes into a region of size 7 overflows the destination [-Werror=stringop-overflow=]
```

Reply:
That's true, Melex, 
sorry for that awful treatment into debug logs.

Now fixed.